android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -231,6 +232,21 @@ public void sendNotification(Bundle bundle) {
                 notification.setVibrate(new long[]{0, vibration});
             }
 
+            int badge = -1;
+            if (bundle.containsKey("badge")) {
```

Comment:
...however in this project, if you take a look at [index.js](https://github.com/zo0r/react-native-push-notification/blob/master/index.js#L121) you can see that the `number` field seems to get translated into `applicationIconBadgeNumber` for iOS.